app/src/main/java/com/github/se/orator/ui/navigation/TopNavigationMenu.kt

```diff
@@ -0,0 +1,37 @@
+package com.github.se.orator.ui.navigation
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.statusBarsPadding
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.platform.testTag
+import com.github.se.orator.ui.theme.AppDimensions
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TopNavigationMenu(
+    title: @Composable () -> Unit = {},
+    actions: @Composable() (RowScope.() -> Unit) = {},
+    navigationIcon: @Composable () -> Unit = {},
+    testTag: String = "top_app_bar"
+) {
+  Column {
+    CenterAlignedTopAppBar(
+        modifier =
+            Modifier.fillMaxWidth()
+                .statusBarsPadding()
+                .shadow(elevation = AppDimensions.elevationSmall)
+                .testTag(testTag),
+        title = title,
+        actions = actions,
+        navigationIcon = navigationIcon)
+    HorizontalDivider(color = MaterialTheme.colorScheme.onSurface)
+  }
+}
```

Comment:
This is a great implementation that helps avoid code duplication. However, I think it could be optimized even more by integrating the Text directly into the `TopNavigationMenu` function as a parameter for the title section.

Since the title is almost always a Text with the same parameters (like color, style, etc.), you could:

- Add a String parameter (e.g., titleText) to pass the title directly aswell as the testTag
- Include optional parameters for color, and style with default values to reduce repetitive code but still be able to customize when needed

Reply:
Great idea, I'll do that. It will also remove the redundant code of the title in the comment above.